coverage/src/main/kotlin/org/rust/coverage/GrcovRunner.kt

```diff
@@ -52,16 +51,13 @@ class GrcovRunner : RsDefaultProgramRunnerBase() {
     }
 
     override fun execute(environment: ExecutionEnvironment) {
-        val state = environment.state as CargoRunStateBase
-
         if (checkNeedInstallGrcov(environment.project)) return
 
-        state.addCommandLinePatch(cargoCoveragePatch)
-        environment.cargoPatches.add(cargoCoveragePatch)
-
+        val state = environment.state as CargoRunStateBase
         val workingDirectory = state.commandLine.workingDirectory.toFile()
         cleanOldCoverageData(workingDirectory)
 
+        environment.cargoPatches.add(cargoCoveragePatch)
```

Comment:
It's quite hard to understand from what threads we modify `cargoPatches` list. And even if we modify them from EDT only currently, nothing stops us to start modify them from another thread in future that may lead to unexpected errors.
I'm not sure that even in this case it will lead to errors but I don't want to care about it a all.
So let's convert `MutableList<CargoPatch>` into some immutable structure and create a new object instead of modifying the old one. Note, user data is already thread-safe because it uses immutable map under the hood